sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnHarness.java

```diff
@@ -115,43 +124,46 @@ public static void main(PipelineOptions options,
   }
 
   public static void main(
+      String id,
       PipelineOptions options,
       Endpoints.ApiServiceDescriptor loggingApiServiceDescriptor,
       Endpoints.ApiServiceDescriptor controlApiServiceDescriptor,
       ManagedChannelFactory channelFactory,
       StreamObserverFactory streamObserverFactory) {
     IdGenerator idGenerator = IdGenerators.decrementingLongs();
-    try (BeamFnLoggingClient logging = new BeamFnLoggingClient(
-        options,
-        loggingApiServiceDescriptor,
-        channelFactory::forDescriptor)) {
+    try (BeamFnLoggingClient logging =
```

Comment:
logging is not used. Shall we remove it?

Reply:
No. This is done to close the client automatically at the completion of the `try` block, and the client must exist to send intercepted LOG messages to the logging server.

Reply:
Can we add a comment explaining this as logging is unused which makes this code block non intuitive.